functions/exercises.py

```diff
@@ -334,7 +389,9 @@ def squared_nums(without_odds=False):
     :return:
     """
     # INSERT YOUR CODE HERE
-    #
+    if without_odds:
+        return [x**2 for x in range(1,31) if x**2 % 2 == 0]
+    return [x**2 for x in range(1,31)]

```

Comment:
```suggestion
    squared = [x**2 for x in range(1,31)]
    if without_odds:
        return [x for x in squared if x% 2 == 0]
    return squared
```

Reducing code duplication

Reply:
With your implementation there isnt code duplication but its less efficient in case without_odds=True